src/Symfony/Component/Yaml/Inline.php

```diff
@@ -659,9 +661,22 @@ private static function evaluateScalar($scalar, $flags, $references = array())
 
                         return $time;
                 }
-            default:
-                return (string) $scalar;
         }
+
+        if (null !== self::$tagResolver && $scalar && '!' === $scalar[0]) {
```

Comment:
I am not a particular big fan of putting conditions like '&& $scalar' because it takes away some readability. There are a lot of ways of validating '$scalar' (empty(), is_null(), etc...) instead of making me believe (at first sight) that it is a boolean.